Bottom: b325db5cbd6588f2d9a334d1cd2f3151b78f6d13
Top:    59164e5ae04b86baa7d0ad25c5b8d95bf440d9f0
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2008-10-02 16:09:37 +0100

Change scroll-wheel bindings to match more GTK applications



---

diff --git a/src/hid/gtk/gui-output-events.c b/src/hid/gtk/gui-output-events.c
index aff3e1d..f033f0f 100644
--- a/src/hid/gtk/gui-output-events.c
+++ b/src/hid/gtk/gui-output-events.c
@@ -1202,18 +1202,18 @@ ghid_port_window_mouse_scroll_cb (GtkWidget * widget,
 
   state = (GdkModifierType) (ev->state);
   mk = ghid_modifier_keys_state (&state);
+
   if (mk == NONE_PRESSED)
+    dy = ghid_port.height * gport->zoom / 40;
+  else if (mk == SHIFT_PRESSED)
+    dx = ghid_port.width * gport->zoom / 40;
+  else if (mk == CONTROL_PRESSED)
     {
       zoom_factor = (ev->direction == GDK_SCROLL_UP) ? 0.8 : 1.25;
       ghid_port_ranges_zoom (gport->zoom * zoom_factor);
       return TRUE;
     }
 
-  if (mk == SHIFT_PRESSED)
-    dy = ghid_port.height * gport->zoom / 40;
-  else
-    dx = ghid_port.width * gport->zoom / 40;
-
   if (ev->direction == GDK_SCROLL_UP)
     {
       dx = -dx;
